src/planner/partialize.c

```diff
@@ -305,21 +305,26 @@ copy_append_like_path(PlannerInfo *root, Path *path, List *new_subpaths, PathTar
 	if (IsA(path, AppendPath))
 	{
 		AppendPath *append_path = castNode(AppendPath, path);
-		append_path->path.pathtarget = pathtarget;
-		return (Path *) copy_append_path(append_path, new_subpaths);
+		AppendPath *new_append_path = copy_append_path(append_path, new_subpaths);
+		new_append_path->path.pathtarget = pathtarget;
```

Comment:
It is a little unclear if this is necessary, but it might be a good idea to copy the `pathtarget` list.

Also consider if you should move this into `copy_append_path` (and the same for the other functions) since it seems to make more sense to keep all of the copy logic in the same place.

```suggestion
		new_append_path->path.pathtarget = copy_pathtarget(pathtarget);
```

Reply:
Changed as suggested